### DISCUSSION TITLE
Why does `Async::HTTP::Server#run` require an `Async::Task`, but `Async::DNS::Server#run` does not?

### DISCUSSION THREAD

**Discussion by postmodern** [socketry/async-http#115] [Q&A]
I noticed that `Async::HTTP::Server#run` must be called within an existing `Async` task, but `Async::DNS::Server#run` always runs in a new `Async::Reactor`. Is this intentional?

**Comment by postmodern** [comment]
I also noticed that `Async::HTTP::Server#run` does not have any default logging messages, while `Async::DNS::Server#run` does.

**Reply by ioquatix** [reply]
The two code bases were created years and years apart so there are differences. Probably `Async::DNS::Server` was more noisy because I felt like logging was more important at the time. i.e. I wanted to see it working.

I think we should find alignment between the two.
